src/PactNet.AspNetCore/Messaging/MessageMiddleware.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using PactNet.AspNetCore.Messaging.Models;
+using PactNet.AspNetCore.Messaging.Options;
+using PactNet.Native;
+
+namespace PactNet.AspNetCore.Messaging
```

Comment:
The more I've been playing around with this locally the more I'm thinking we should provide an entire server instead of only the middleware portion. It puts a lot of onus on the user to set up their own ASP.Net Core host and configure it, and all we need is a fixed endpoint so the Rust verifier can call it.

I'm not sure users will really want to do all that work of copying over all of our boilerplate, and I'm not sure it really gives any advantage to them either. Can you think of any reasons this is better than providing a single turnkey solution?

We could abstract away the entire implementation details from the user and they wouldn't even need to know a REST API had been started at all. It does feel really odd at first glance that you have to create an entire REST API to verify pub/sub messages, especially if the thing you are testing doesn't even have a REST API (like if it's a Windows service or console app that just subscribes to a queue and processes messages).

I think we could do all of that for the user instead so that they don't have to, and they'd never even need to know that there is a REST API running at all. That just becomes one of our implementation details that's hidden away.

Reply:
I believe that is how the java have done it.
That's how we've implemented it in our lib. We actually never build any service. We just register the scenarios that are "pushed" in a fake way and add it to the invokers. It's just code testing at the end.